x/permission/keeper/keeper.go

```diff
@@ -71,9 +72,37 @@ func (k Keeper) AddGroupMember(ctx sdk.Context, groupID math.Uint, member sdk.Ac
 	id := k.groupMemberSeq.NextVal(store)
 	store.Set(memberKey, id.Bytes())
 	store.Set(types.GetGroupMemberByIDKey(id), k.cdc.MustMarshal(&groupMember))
+	return id, nil
+}
+
+func (k Keeper) AddGroupMemberWithExpiration(ctx sdk.Context, groupID math.Uint, member sdk.AccAddress, expiration time.Time) error {
+	id, err := k.AddGroupMember(ctx, groupID, member)
+	if err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	// We can simply override the whole value here, because the expiration time is the only field in the value.
+	// If there are more fields in the future, we should use a more sophisticated way to update the value.
+	memberExtra := types.GroupMemberExtra{
+		ExpirationTime: expiration,
+	}
+	store.Set(types.GetGroupMemberExtraKey(groupID, member), id.Bytes())
```

Comment:
Why we have to set two storage here, is it possible to reduce to one?

Reply:
yes, merge it to `GroupMember`